lib/active_type.rb

```diff
@@ -1,14 +1,31 @@
 # encoding: utf-8
 
 require 'active_type/version'
-require 'active_record'
-require 'active_type/util'
-require 'active_type/record'
-require 'active_type/object'
-
-if ActiveRecord::VERSION::STRING == '4.2.0'
-  raise(<<-MESSAGE.strip_heredoc)
-    ActiveType is not compatible with ActiveRecord 4.2.0. Please upgrade to 4.2.1
-    For details see https://github.com/makandra/active_type/issues/31
-  MESSAGE
+
+
+load_now = proc do
+  require 'active_type/util'
+  require 'active_type/record'
+  require 'active_type/object'
+
+  if ActiveRecord::VERSION::STRING == '4.2.0'
+    raise(<<-MESSAGE.strip_heredoc)
+      ActiveType is not compatible with ActiveRecord 4.2.0. Please upgrade to 4.2.1
+      For details see https://github.com/makandra/active_type/issues/31
+    MESSAGE
+  end
+end
+
+
+if defined?(Rails) && defined?(ActiveSupport)
```

Comment:
I suspect the check is unnecessary anways, since you probably cannot have `Rails` but not `ActiveSupport`. This was more paranoia.

I'm not sure what other gems do. I've just seen the Devise solution, but that's a different case, since Devise does not work without Rails anyways. I don't want users to have to require `active_record` manually in case they don't use Rails.